api/pkg/cmd/stack.go

```diff
@@ -44,18 +49,43 @@ func (s Stack) GetAppStacks(ctx context.Context, payload model.AppStackPayload)
 	return nil, nil
 }
 
-func convertParamToStacklist(paramOutput string, payload model.AppStackPayload) ([]*model.AppStack, error) {
+func parseParamToStacklist(paramOutput string) ([]string, error) {
 	var stacklist []string
 	err := json.Unmarshal([]byte(paramOutput), &stacklist)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not parse json")
-	}
+	return stacklist, errors.Wrap(err, "could not parse json")
+}
+
+func enrichStacklistMetadata(ctx context.Context, stacklist []string, payload model.AppStackPayload, integrationSecret *config.IntegrationSecret) ([]*model.AppStackResponse, error) {
+	workspaceRepo := workspace_repo.NewWorkspaceRepo(
+		integrationSecret.Tfe.Url,
+		integrationSecret.Tfe.Org,
+	).WithTFEToken(setup.GetConfiguration().TFE.Token)
+	wg := sync.WaitGroup{}
+
+	stacks := []*model.AppStackResponse{}
 
-	stacks := []*model.AppStack{}
 	for _, stackName := range stacklist {
-		appStack := model.MakeAppStack(payload.AppName, payload.Environment, stackName)
-		stacks = append(stacks, &appStack)
+		wg.Add(1)
+		go func(stackName string) {
+			stack := &model.AppStackResponse{
+				AppMetadata: *model.NewAppMetadata(payload.AppName, payload.Environment, stackName),
+			}
+			workspace, err := workspaceRepo.GetWorkspace(ctx, fmt.Sprintf("%s-%s", payload.AppMetadata.Environment, stackName))
+			if err == nil {
```

Comment:
rather than nesting, fail fast for error cases

```suggestion
			if err != nil {
			   return err
			}
```

Reply:
see also "golang.org/x/sync/errgroup". Might be helpful for capturing these errors a little better

Reply:
This `if err == nil {` structure is intended in this case. It should just omit the workspace values when it can't find a workspace for a stack, similar to how the cli operates

I did make a small update to the behavior though, I added an error field on the response object that will be populated with the error when it's not nil